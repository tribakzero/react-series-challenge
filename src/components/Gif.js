import React, { Component } from 'react';
import LikeStatus from './LikeStatus';

class Gif extends Component {
  render(){
    return (
      <div className="gif">
        <img src={this.props.src} />
        <LikeStatus liked={this.props.liked} />
      </div>
    )  
  }
}

export default Gif;
