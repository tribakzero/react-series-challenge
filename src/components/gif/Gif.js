import React, { Component } from 'react';
import { connect } from 'react-redux';
import LikeStatus from 'components/likeStatus';

export class Gif extends Component {
  constructor(props) {
    super(props);
  }
  render() {
    let likeRender = <LikeStatus liked={false} />;
    if (
      this.props.favorite &&
      this.props.favorite[this.props.id] &&
      Boolean(this.props.favorite[this.props.id].liked)
    ) {
      likeRender = <LikeStatus liked={true} />;
    }
    return (
      <div className="gif" src={this.props.src} onClick={this.props.onClick}>
        <img
          className="img"
          src={this.props.src}
          alt={this.props.title}
          title={this.props.title}
        />
        { likeRender }
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  const {
    favorite
  } = state;
  
  return {
    favorite
  };
};

export default connect(mapStateToProps)(Gif);
