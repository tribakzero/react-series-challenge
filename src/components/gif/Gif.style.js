import styled from 'styled-components';


export const StyledGif = styled.div.attrs({
  style: props => ({
    background: props.bgColor(),
  })
})`
  width: auto;
  margin: 1em;
  display: flex;
  min-width: 66px;
  position: relative;
  .like-status {
    position: absolute;
    visibility: hidden;
  }
  &:hover .like-status {
    visibility: visible;
  }
`