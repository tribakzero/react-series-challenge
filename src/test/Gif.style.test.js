import React from 'react';
import { shallow } from 'enzyme';
import { Gif } from 'components/gif/Gif'
import 'jest-styled-components'

describe('Gif', () => {
  test('renders properly', () => {
    const component = shallow(<Gif />);
    expect(component).toMatchSnapshot()
  })
});
