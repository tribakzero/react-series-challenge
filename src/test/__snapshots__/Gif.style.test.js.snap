// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gif renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bgColor": [Function],
      "children": Array [
        <img
          alt={undefined}
          className="img"
          src={undefined}
          title={undefined}
/>,
        <LikeStatus
          liked={false}
/>,
      ],
      "onClick": undefined,
      "src": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": undefined,
          "className": "img",
          "src": undefined,
          "title": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "liked": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bgColor": [Function],
        "children": Array [
          <img
            alt={undefined}
            className="img"
            src={undefined}
            title={undefined}
/>,
          <LikeStatus
            liked={false}
/>,
        ],
        "onClick": undefined,
        "src": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": undefined,
            "className": "img",
            "src": undefined,
            "title": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "liked": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
